compiler/noirc_evaluator/src/acir/generated_acir.rs

```diff
@@ -582,7 +590,7 @@ impl<F: AcirField> GeneratedAcir<F> {
         if num_bits >= F::max_num_bits() {
             return Err(RuntimeError::InvalidRangeConstraint {
                 num_bits: F::max_num_bits(),
-                call_stack: self.call_stack.clone(),
+                call_stack: self.get_call_stack(),
```

Comment:
```suggestion
                call_stack: self.call_stacks.get_call_stack(self.call_stack_id),
```
Is this not valid and then we can get rid of `call_stack: CallStack`?

Reply:
Yes it is valid, and it does indeed simplify the code a little bit since we can get rid of self.call_stack.
However, it will force to recompute the call stack every time, even if the source location has not changed.
Since this optimisation is very simple, I figure we should keep it.

Reply:
I feel recomputing the call stack should not take much time, especially as this error should be rare. But as this PR has been up very long I am not going to block on this.